DNN Platform/Library/Data/DataProvider.cs

```diff
@@ -3353,6 +3354,14 @@ public virtual DateTime GetDatabaseTime()
             return ExecuteScalar<DateTime>("GetDatabaseTime");
         }
 
+        public virtual DateTimeOffset GetDatabaseTimeOffset() {
+            var dateTimeString = ExecuteSQL<DateTimeOffset>("SELECT SYSDATETIMEOFFSET()").ToString();
+            DateTimeOffset dateTimeOffset;
+            DateTimeOffset.TryParse(dateTimeString, out dateTimeOffset);
```

Comment:
You initially get a `DateTimeOffset` only to stringify and parse again. Is that necessary?